### DISCUSSION TITLE
How to build a dual CLI/Interactive Menu

### DISCUSSION THREAD

**Discussion by gtapia-cuscal** [spectreconsole/spectre.console#1610] [Q&A]
I want to build a standard cli with regualr cli commands like `git branch create ...` vs `git` to bring up an interactive menu.  To do this I am finding that I have to hack the DI, create a redundat tree structure, etc.  Is there any example or best practices to do this? 

So if you have a complete command argument list, then use the standard CLI to execute the command.  If just the root executable command (`git`) is run then revert to an interactive menu.  I want the interactive menu to leverage the infrastructure (branches, commands, settings, etc) that has been configured for the regular cli.

Obviously, the interactive menu will have additional methods to build up the `Settings` object by promtpting for input, etc.  Any ideas?

**Comment by FrankRay78** [comment]
Interestingly, an issue has been recently raised for just this, see: https://github.com/spectreconsole/spectre.console/issues/1654

**Comment by Panda-Sharp** [comment]
I was struggling with something really similar, so I came up with this solution:
https://github.com/spectreconsole/spectre.console/discussions/2206

(Any honest feedback, is really appreciated)
